## Used Libariries. 

## Import Data

In [2]:
import os
# Get the current working directory
current_directory = os.getcwd()
print("Current Working Directory:", current_directory)

Current Working Directory: C:\Users\jackf\OneDrive\Desktop\Bike_Sharing\UCI_Bike_Sharing


In [3]:
import pandas as pd

# Specify the path to your CSV file including the file name
file_path = r'C:\Users\jackf\OneDrive\Desktop\Bike_Sharing\UCI_Bike_Sharing\data\day.csv'

# Read the CSV file into a DataFrame
day_Bike = pd.read_csv(file_path)

# Display the DataFrame
day_Bike

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
0          1  2011-01-01       1   0     1        0        6           0   
1          2  2011-01-02       1   0     1        0        0           0   
2          3  2011-01-03       1   0     1        0        1           1   
3          4  2011-01-04       1   0     1        0        2           1   
4          5  2011-01-05       1   0     1        0        3           1   
..       ...         ...     ...  ..   ...      ...      ...         ...   
726      727  2012-12-27       1   1    12        0        4           1   
727      728  2012-12-28       1   1    12        0        5           1   
728      729  2012-12-29       1   1    12        0        6           0   
729      730  2012-12-30       1   1    12        0        0           0   
730      731  2012-12-31       1   1    12        0        1           1   

     weathersit      temp     atemp       hum  windspeed  casual  registered  \
0             2  0.344167  0.363625  0.805833   0.160446     331         654   
1             2  0.363478  0.353739  0.696087   0.248539     131         670   
2             1  0.196364  0.189405  0.437273   0.248309     120        1229   
3             1  0.200000  0.212122  0.590435   0.160296     108        1454   
4             1  0.226957  0.229270  0.436957   0.186900      82        1518   
..          ...       ...       ...       ...        ...     ...         ...   
726           2  0.254167  0.226642  0.652917   0.350133     247        1867   
727           2  0.253333  0.255046  0.590000   0.155471     644        2451   
728           2  0.253333  0.242400  0.752917   0.124383     159        1182   
729           1  0.255833  0.231700  0.483333   0.350754     364        1432   
730           2  0.215833  0.223487  0.577500   0.154846     439        2290   

      cnt  
0     985  
1     801  
2    1349  
3    1562  
4    1600  
..    ...  
726  2114  
727  3095  
728  1341  
729  1796  
730  2729  

[731 rows x 16 columns]

## Linear Regression Model

**Note:**: It should be noted for us to build the model correctly, we will have to drop the columns "cnt", "casual", and "registered", as they will most likely break the code, giving you a value of "1" ajusted R-Squared. The reason is because we do "casual" riders + "registered" riders  = "cnt" of total rider for that given day. So we must drop these columns, "dteday" and "instant" can also be dropped but since they are unique values for each column, no need to drop them, as they will be helpful when building ARIMA model or Random Forest where we clean the data-up. We also do random_state, as this time-series oriented data, for this code to work that is necessary. Also, we do 20/80 split because the data isn't very large, its only 731 datapoints.

In [11]:
import pandas as pd
from sklearn.model_selection import train_test_split
import statsmodels.formula.api as smf

X = day_Bike.drop(['cnt', "casual", "registered"], axis=1)
y = day_Bike['cnt']

# Split the data into training and testing sets (adjust the test_size parameter as needed)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

formula = ('cnt ~ season + yr + mnth + holiday + weekday + '
           'workingday + weathersit + temp + atemp + hum + windspeed')

ols_model = smf.ols(formula=formula, data=day_Bike)

model1 = ols_model.fit()
print(model1.summary())

                            OLS Regression Results                            
Dep. Variable:                    cnt   R-squared:                       0.800
Model:                            OLS   Adj. R-squared:                  0.797
Method:                 Least Squares   F-statistic:                     261.9
Date:                Sat, 02 Dec 2023   Prob (F-statistic):          7.80e-243
Time:                        04:12:00   Log-Likelihood:                -5981.0
No. Observations:                 731   AIC:                         1.199e+04
Df Residuals:                     719   BIC:                         1.204e+04
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1469.0031    240.218      6.115      0.0

In [12]:
from sklearn.metrics import r2_score
y_pred = model1.predict(X_test)

r_squared = r2_score(y_test, y_pred)
print(f'R-squared on the test set: {r_squared}') 

R-squared on the test set: 0.8324560523959417
